### PR TITLE
IFTTT Webhooks Notification

### PR DESCRIPTION
**Pull request by lcdtyph**
Thanks for this amazing project!

I've simply implemented the ifttt webhooks notify hook.

This hook needs `IFTTT_API_KEY` and `IFTTT_EVENT_NAME`. The api key can be obtained at [https://ifttt.com/maker_webhooks](https://ifttt.com/maker_webhooks), and the event name can be set while creating an ifttt applet with webhooks.

This notification method could be a bridge to make acme.sh work with tons of IFTTT services.

### CODE REVIEW

**Comment by Neilpang** [comment]
please update the usage here:
https://github.com/Neilpang/acme.sh/wiki/notify

File: notify/ifttt.sh [hunk 1]
```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env bash
```
Comment (Neilpang) [inline notify/ifttt.sh]: use `sh`  instead of `bash`

Comment (lcdtyph) [reply notify/ifttt.sh]: Done, thanks!

**Comment by lcdtyph** [comment]
Should I open an issue for bug reporting?

**Comment by Neilpang** [comment]
yes

File: notify/ifttt.sh [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    return 1
+  fi
+
+  if [ "$IFTTT_SUBJECT_KEY" = "$IFTTT_CONTENT_KEY" ]; then
+    IFTTT_SUBJECT_KEY=""
+    IFTTT_CONTENT_KEY=""
+    _err "The IFTTT_SUBJECT_KEY must not be same as IFTTT_CONTENT_KEY."
+    return 1
+  fi
+  _saveaccountconf_mutable IFTTT_SUBJECT_KEY "$IFTTT_SUBJECT_KEY"
```
Comment (Neilpang) [inline notify/ifttt.sh]: don't save the values here.

we should only save the values when the users set a value.

we should not save it if it was empty but used the default value.

**Comment by lcdtyph** [comment]
I've created the issue, https://github.com/Neilpang/acme.sh/issues/2421, and appended to the wiki already.

File: notify/ifttt.sh [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+  _IFTTT_CONTENT_KEY="${IFTTT_CONTENT_KEY:-$(_readaccountconf_mutable IFTTT_CONTENT_KEY)}"
+  IFTTT_CONTENT_KEY="$_IFTTT_CONTENT_KEY"
+  if [ -z "$IFTTT_CONTENT_KEY" ]; then
+    IFTTT_CONTENT_KEY="value2"
+    _info "The IFTTT_CONTENT_KEY is not set, so use the default value2 as key."
+  elif ! _hasfield "$_IFTTT_AVAIL_MSG_KEYS" "$IFTTT_CONTENT_KEY"; then
+    _err "The IFTTT_CONTENT_KEY \"$IFTTT_CONTENT_KEY\" is not available, should be one of $_IFTTT_AVAIL_MSG_KEYS"
+    IFTTT_CONTENT_KEY=""
+    return 1
```
Comment (Neilpang) [inline notify/ifttt.sh]: add  here
```
else
    _saveaccountconf_mutable xxxxxxxx
```

File: notify/ifttt.sh [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+  fi
+
+  if [ "$IFTTT_SUBJECT_KEY" = "$IFTTT_CONTENT_KEY" ]; then
+    IFTTT_SUBJECT_KEY=""
+    IFTTT_CONTENT_KEY=""
+    _err "The IFTTT_SUBJECT_KEY must not be same as IFTTT_CONTENT_KEY."
+    return 1
+  fi
+
+  if [ -n "$_IFTTT_SUBJECT_KEY" ]; then
```
Comment (Neilpang) [inline notify/ifttt.sh]: no

File: notify/ifttt.sh (hunk 3 above)
Comment (lcdtyph) [reply notify/ifttt.sh]: I was worrying about saving incorrect values, so I thought it's better to save them after all checks.

Comment (Neilpang) [reply notify/ifttt.sh]: if the value is set by the user,  just save it.

if it's from a default value,   don't save it.

**Pull request merged**